src/hotspot/share/utilities/tribool.hpp

```diff
@@ -40,11 +40,15 @@ class TriBool{
 
  public:
   TriBool() : _value(0) {}
-  TriBool(bool value) : _value(((u1)value) | 2) {}
+  TriBool(bool value) : _value(value) {
+    // set to not-default in separate step to avoid conversion warnings
+    _value |= 2;
```

Comment:
```suggestion
  TriBool(bool value) {
    *this = value;
```

Reply:
Thanks for the suggestion but this makes me squint to try to figure out what it does and why it's correct.